### DISCUSSION TITLE
How to use Lucide icon library?

### DISCUSSION THREAD

**Discussion by jimmiejackson414** [callstack/react-native-paper#4500] [Q&A]
Has anybody had any success in getting Lucide to work with RNP? I know the docs mention passing the icons to the settings prop for the PaperProvider, but I can't seem to find what exactly to import? And does doing this now allow me to use the string names of the lucide icons, as if it were using the default Material Community Icons?

**Comment by alexislouie** [comment]
A bit late but in case anyone else runs into this issue, here is what worked for me.

You want to override the Paper's default icon settings with lucide icons as you mentioned, but you will also need to convert lucide icons' props to match RNP's icons' prop shape. 

```
import {icons as lucideIcons} from 'lucide-react-native';
import {IconProps} from 'react-native-paper/lib/typescript/components/MaterialCommunityIcon';

const LucideIcon = ({...props}: IconProps) => {
  const Icon = lucideIcons[props.name as keyof typeof lucideIcons];
  return Icon ? <Icon {...props} /> : null;
};

export default function RootLayout() {
  return (
    <PaperProvider
      settings={{
        icon: (props) => <LucideIcon {...props} />,
      }}>
      // content here
    </PaperProvider>
  );
}
```

Lucide Icon names are capitalized so when you're using the icons, you'll want to use them like this: 
```
<Chip
  icon="Camera"
  mode="outlined">
  Example
</Chip>

<Icon source="Scan" size={20} color={theme.colors.onPrimary} />
```

Or you can update the `LucideIcon` above to handle the lowercase names and convert them to pascal case.
